addon-test-support/@ember/test-helpers/-internal/debug-info.ts

```diff
@@ -0,0 +1,169 @@
+import {
+  run,
+  DebugInfo as BackburnerDebugInfo,
+  QueueItem,
+  DeferredActionQueues,
+} from '@ember/runloop';
+import { assign } from '@ember/polyfills';
+
+const PENDING_AJAX_REQUESTS = 'Pending AJAX requests';
+const PENDING_TEST_WAITERS = 'Pending test waiters';
+const SCHEDULED_ASYNC = 'Scheduled async';
+const SCHEDULED_AUTORUN = 'Scheduled autorun';
+
+type MaybeDebugInfo = BackburnerDebugInfo | null;
+
+interface ISettledState {
+  hasPendingTimers: boolean;
+  hasRunLoop: boolean;
+  hasPendingWaiters: boolean;
+  hasPendingRequests: boolean;
+}
+
+interface ISummaryInfo {
+  hasPendingRequests: boolean;
+  hasPendingWaiters: boolean;
+  autorunStackTrace: string | undefined | null;
+  pendingTimersCount: number;
+  hasPendingTimers: boolean;
+  pendingTimersStackTraces: (string | undefined)[];
+  pendingScheduledQueueItemCount: Number;
+  pendingScheduledQueueItemStackTraces: (string | undefined)[];
+  hasRunLoop: boolean;
+}
+
+/**
+ * Retrieves debug information from backburner's current deferred actions queue (runloop instance).
+ * If the `getDebugInfo` method isn't available, it returns `null`.
+ *
+ * @public
+ * @returns {MaybeDebugInfo | null} Backburner debugInfo or, if the getDebugInfo method is not present, null
+ */
+export function getDebugInfo(): MaybeDebugInfo {
+  let debugEnabled = run.backburner.DEBUG === true;
+  let getDebugInfoAvailable = typeof run.backburner.getDebugInfo === 'function';
+
+  return debugEnabled && getDebugInfoAvailable
+    ? <BackburnerDebugInfo>run.backburner.getDebugInfo()
+    : null;
+}
+
+/**
+ * Encapsulates debug information for an individual test. Aggregates information
+ * from:
+ * - info provided by getSettledState
+ *    - hasPendingTimers
+ *    - hasRunLoop
+ *    - hasPendingWaiters
+ *    - hasPendingRequests
+ * - info provided by backburner's getDebugInfo method (timers, schedules, and stack trace info)
+ *
+ */
+export default class DebugInfo {
```

Comment:
Let’s make an IDebugInfo interface, which is public, then make this class implement the interface. Specifically, I want to make it clear that _using_ the public members of IDebugInfo is good, but relying on internal class implementation details is not. Also, it makes it clearer that the user can’t instantiate per-se (that you’d get one by way of getSettledState).

Reply:
Agree. Doing it.

Reply:
I'm doing some renaming here for consistency and clarity.